src/hooks/useVenueAccessToken.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect } from "react";
+import firebase from "firebase/app";
+
+import { AnyVenue } from "types/Firestore";
+
+import { WithId } from "utils/id";
+import { accessTokenKey } from "utils/localStorage";
+import { isTruthy, notEmpty } from "utils/types";
+import { checkAccess, getLocalStorageToken } from "functions/auth";
+
+export const useVenueAccessToken = (
+  venue?: WithId<AnyVenue>,
+  onDenyAccess?: () => void
+) => {
+  useEffect(() => {
+    if (!venue) return;
+
+    const denyAccess = () => {
+      localStorage.removeItem(accessTokenKey(venue.id));
+      onDenyAccess && onDenyAccess();
+    };
+
+    if (notEmpty(venue.access)) {
+      console.log("venue.access", venue.access);
+      const token = getLocalStorageToken(venue.id) ?? undefined;
+      console.log("found token:", token);
+      if (!token) {
+        console.log("checking access");
+        checkAccess({
+          venueId: venue.id,
+          token,
+        }).then((result) => {
+          console.log(
+            "access check result:",
+            result,
+            "isTruthy result.data:",
+            isTruthy(result.data)
+          );
+          if (!isTruthy(result.data)) {
+            firebase
+              .auth()
+              .signOut()
+              .finally(() => {
+                denyAccess();
+              });
+          }
+        });
+      } else {
+        firebase
+          .auth()
+          .signOut()
+          .finally(() => {
+            denyAccess();
+          });
```

Comment:
Maybe move this to a guard statement near the top?

Not sure if the helper exists as written here, treat it as pseudocode:

```typescript
if (isEmpty(venue.access)) {
  firebase
          .auth()
          .signOut()
          .finally(() => {
            denyAccess();
          });
  return
}
```

Reply:
Looking at the logic in this function more.. I also wonder if it's correct?

It seems to be signing out the user if `venue.access` is empty, which seems wrong.

My guess is the intent was that it should sign out the user doesn't have the correct venue access rights.

Even in that case, I'm not sure that we would want to sign them out entirely though.. possibly just show them an 'access denied' or similar?

eg. If I have access to the main map of the party, but not one of the subvenues, I want to be able to have the platform to show me some kind of useful 'denied' message, but still let me access the other locations that I am allowed to go.

Reply:
Not true. This chunk of code is being executed when `token`  exists.

Reply:
```jsx

if (!token) {
        checkAccess({
          venueId: venue.id,
          token,
        })
      } else {
        firebase
          .auth()
          .signOut()
          .finally(() => {
            denyAccess();
          });     
}
```

Reply:
Looks like the logic needs to be reverse. and the if statement should be `if (token)`

Reply:
Fair enough. That wasn't clear from the code, might have been the way it was nested that was confusing.